### ISSUE TITLE
Auto-Detection mode produces different result for the same file

### ISSUE THREAD

**Issue by gone-for-coding** [ansible/ansible-lint#2164]
##### Summary

When running `ansible-lint` in a directory with several files and one playbook only one issue is reported whereas `ansible-lint filename` also reports further issues in the file.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.12.4]
  config file = /home/sewert/ansible-gitlab-runner-image/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.13 (default, May 11 2022, 08:30:17) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.0.2 using ansible 2.12.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Debian 11

##### STEPS TO REPRODUCE

Create a playbook with several issues and test as stated above.

```console (paste below)
---
- hosts: localhost
  name: test
  gather_facts: False
  tasks:
    - name: Check pip packages
      ansible.builtin.pip:
        name:
          - ansible-core==2.12.4
          - ansible-lint-junit==0.17.4
      check_mode: yes
        # bad comment

    - shell: whoami
```

##### Desired Behavior

yaml: comment not indented like content (comments-indentation)
install_check.yml:48

command-instead-of-shell: Use shell only when shell functionality is required.
install_check.yml:50 Task/Handler: shell  whoami

fqcn-builtins: Use FQCN for builtin actions.
install_check.yml:50 Task/Handler: shell  whoami

no-changed-when: Commands should not change things if nothing needs doing.
install_check.yml:50 Task/Handler: shell  whoami

unnamed-task: All tasks should be named.
install_check.yml:50 Task/Handler: shell  whoami

##### Actual Behavior

yaml: comment not indented like content (comments-indentation)
install_check.yml:48

##### Verbose output of both tests

```console
# ansible-lint -vvvv
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/root/.cache/ansible-compat/454622', colored=True, config_file=None, configured=True, cwd=PosixPath('/home/sewert/ansible-gitlab-runner-image'), display_relative_path=True, enable_list=[], exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], extra_vars=None, format='rich', kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'text': '**/templates/**/*.*'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], lintables=[], listrules=False, listtags=False, loop_var_prefix=None, mock_modules=[], mock_roles=[], offline=None, parseable=False, progressive=False, project_dir='.', quiet=0, rules={}, rulesdir=[], rulesdirs=['/usr/local/lib/python3.8/site-packages/ansiblelint/rules'], skip_action_validation=True, skip_list=[], tags=[], use_default_rules=False, var_naming_pattern=None, verbosity=4, version=False, warn_list=['experimental', 'role-name'], write=False)
DEBUG    /home/sewert/ansible-gitlab-runner-image
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/454622/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/454622/collections:/usr/share/ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/454622/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Loading rules from /usr/local/lib/python3.8/site-packages/ansiblelint/rules
WARNING  Loading custom .yamllint config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 6, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': {'level': 'warning'}, 'document-end': False, 'document-start': {'level': 'warning', 'present': True}, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'warning', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false', 'True', 'False', 'yes', 'no', 'Yes', 'No'], 'check-keys': True}}
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
DEBUG    Excluded .gitlab-ci.yml (yaml)
DEBUG    Examining .yamllint of type yaml
DEBUG    Examining install_check.yml of type yaml
DEBUG    Examining .pre-commit-config.yaml of type yaml
WARNING  Listing 1 violation(s) that are fatal
yaml: comment not indented like content (comments-indentation)
install_check.yml:48

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - yaml  # Violations reported by yamllint.

Finished with 1 failure(s), 0 warning(s) on 35 files.

# ansible-lint -vvvv install_check.yml
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/root/.cache/ansible-compat/454622', colored=True, config_file=None, configured=True, cwd=PosixPath('/home/sewert/ansible-gitlab-runner-image'), display_relative_path=True, enable_list=[], exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], extra_vars=None, format='rich', kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'text': '**/templates/**/*.*'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], lintables=['install_check.yml'], listrules=False, listtags=False, loop_var_prefix=None, mock_modules=[], mock_roles=[], offline=None, parseable=False, progressive=False, project_dir='.', quiet=0, rules={}, rulesdir=[], rulesdirs=['/usr/local/lib/python3.8/site-packages/ansiblelint/rules'], skip_action_validation=True, skip_list=[], tags=[], use_default_rules=False, var_naming_pattern=None, verbosity=4, version=False, warn_list=['experimental', 'role-name'], write=False)
DEBUG    /home/sewert/ansible-gitlab-runner-image
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/454622/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/454622/collections:/usr/share/ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/454622/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Loading rules from /usr/local/lib/python3.8/site-packages/ansiblelint/rules
WARNING  Loading custom .yamllint config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 6, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': {'level': 'warning'}, 'document-end': False, 'document-start': {'level': 'warning', 'present': True}, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'warning', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false', 'True', 'False', 'yes', 'no', 'Yes', 'No'], 'check-keys': True}}
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: install_check.yml
INFO     Executing syntax check on install_check.yml (1.20s)
DEBUG    Examining install_check.yml of type playbook
WARNING  Listing 5 violation(s) that are fatal
yaml: comment not indented like content (comments-indentation)
install_check.yml:48

command-instead-of-shell: Use shell only when shell functionality is required.
install_check.yml:50 Task/Handler: shell  whoami

fqcn-builtins: Use FQCN for builtin actions.
install_check.yml:50 Task/Handler: shell  whoami

no-changed-when: Commands should not change things if nothing needs doing.
install_check.yml:50 Task/Handler: shell  whoami

unnamed-task: All tasks should be named.
install_check.yml:50 Task/Handler: shell  whoami

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - command-instead-of-shell  # Use shell only when shell functionality is required.
  - fqcn-builtins  # Use FQCN for builtin actions.
  - no-changed-when  # Commands should not change things if nothing needs doing.
  - unnamed-task  # All tasks should be named.
  - yaml  # Violations reported by yamllint.

Finished with 5 failure(s), 0 warning(s) on 1 files.
```
